app/models/data_resources/event_record.rb

```diff
@@ -48,17 +51,25 @@ class EventRecord < ApplicationRecord
     timespan_to_search = (start_date..end_date).to_a
 
     list_of_fixed_dates = FixedDate.where.not(date_start: nil).to_a.select do |a|
-      timespan = [] if a.date_start.blank?
```

Comment:
we do not have to check for empty `date_start`s  if we already query for `where.not(date_start: nil)`, right?!

Reply:
`where.not(date_start: nil)` only filters out records where date_start is nil.
`blank? `would also account for empty strings ("") or strings like (" ") etc

Reply:
good point, but that should be part of the query to exclude them as well, right? and not querying data to deselect them afterwards in a loop.

Reply:
exactly, it's more efficient to filter them out in the query itself rather than post-processing in a loop

Reply:
Could you provide a suggestion for an updated query here?

Reply:
just `where.not(date_start: nil)` should be enough i think

Reply:
I agree, if there are other values, something else is wrong and shouldn't be tackled here.